File: docs/javascripts/mermaid-config.js
/**
 * Configurazione Mermaid per Material for MkDocs (supporto nativo)
 * Material legge window.mermaidConfig prima di inizializzare Mermaid.
 * securityLevel 'loose' è necessario per i tag HTML nei nodi (es. <br/>)
 */
window.mermaidConfig = {
    startOnLoad: false,
    theme: 'base',
    securityLevel: 'loose',
    flowchart: {
        htmlLabels: true,
        useMaxWidth: true,
        curve: 'basis'
    }
};

File: docs/_filters/mermaid-config.lua
--- mermaid-config.lua
--- Filtro Lua per iniettare configurazione globale Mermaid in ogni blocco.
--- Aggiunge %%{init}%% con securityLevel 'loose' e htmlLabels per supportare
--- tag HTML (es. <br/>) nei nodi dei diagrammi.

local mermaid_init = '%%{init: {"securityLevel": "loose", "theme": "base", "flowchart": {"htmlLabels": true, "useMaxWidth": true, "curve": "basis"}}}%%'

function CodeBlock(el)
  if el.classes:includes("mermaid") then
    -- Non aggiungere se è già presente una direttiva init
    if not el.text:match("^%%%%{init:") then
      el.text = mermaid_init .. "\n" .. el.text
    end
  end
  return el
end
